lib/internal/bootstrap/pre_execution.js

```diff
@@ -158,6 +165,45 @@ function setupSignalHandlers() {
   process.on('removeListener', stopListeningIfSignal);
 }
 
+
+function setupStacktracePrinterOnSigint() {
+  // Use tty_wrap directly to produce minimal require footprint.
+  const { isTTY } = internalBinding('tty_wrap');
+  if (!isTTY(/** stderr */2)) {
+    return;
+  }
+  const { SigintWatchdog, SignalFlags } = require('internal/watchdog');
+
+  const watchdog = new SigintWatchdog();
+  watchdog.oninterrupt = hideStackFrames(function interrupt(flag) {
+    if (process.listenerCount('SIGINT') > 0) {
+      // Simply ignore this watchdog interrupt to prevent duplicate
+      // handling of SIGINT.
+      return;
+    }
+    const console = require('console');
+
+    // Keyboard interruption may not be captured by uncaughtException
+    // listener.
+    if (flag === SignalFlags.FromIdle) {
+      const stackTraceLimit = Error.stackTraceLimit;
+      Error.stackTraceLimit = 0;
+      const err = new KEYBOARD_INTERRUPT();
+      Error.stackTraceLimit = stackTraceLimit;
+      // util.inspect would wrap no stacktrace error with brackets.f
+      console.error(`${err.name} [${err.code}]: ${err.message}`);
+    } else {
+      const err = new KEYBOARD_INTERRUPT();
```

Comment:
I see no need using the internal error system if this is not going to be captured anyway?